src/Modules/reservationPopUp.js

```diff
@@ -0,0 +1,189 @@
+    throw new Error(error);
+  }
+};
+
+const addReservation = async (itemId, name, dateStart, dateEnd) => {
+  const response = await fetch('https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/zRod3rPxBRjxEDaYzujw/reservations', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      item_id: itemId,
+      username: name,
+      date_start: dateStart,
+      date_end: dateEnd,
+    }),
+  });
+
+  if (!response.ok) {
+    const errorMessage = await response.text();
+    throw new Error(errorMessage);
+  }
+};
+
+const domReservations = () => {
+  const itemsContainer = document.getElementById('comment-list');
+  const childCount = itemsContainer.childElementCount;
+
+  const reservationCounter = document.getElementById('reservation-counter');
+  reservationCounter.innerHTML = `Reservations (${childCount})`;
+};
+
+const reservationPopUp = () => {
+  const reservBtn = Array.from(document.getElementsByClassName('btn-2'));
+  const pop = document.querySelector('.reserv-pop');
+  pop.setAttribute('style', 'display: none;');
+
+  const createPopup = async (data) => {
+    const popup = document.createElement('div');
+    popup.classList.add('popup');
+
+    const closeBtn = document.createElement('div');
+    closeBtn.classList.add('close-btn', 'text-4xl', 'py-2', 'cursor-pointer', 'text-right');
+    closeBtn.innerHTML = '&times;';
+    closeBtn.addEventListener('click', () => {
+      popup.remove();
+      pop.setAttribute('style', 'display: none;');
+    });
+
+    const popupImgContainer = document.createElement('div');
+    popupImgContainer.classList.add('img-container');
+
+    const popupImg = document.createElement('img');
+    popupImg.src = data.images?.coverart || data.images?.background || `https://dummyimage.com/400x400/000/1aff00&text=${data.title}`;
+
+    const popupTitle = document.createElement('h3');
+    popupTitle.classList.add('text-3xl', 'font-bold');
+    popupTitle.innerHTML = data.title;
+
+    const commentsTitle = document.createElement('h3');
+    commentsTitle.id = 'reservation-counter';
+
+    const reservations = getReservation(data.key).then((res) => {
+      if (res.error) {
+        commentsTitle.textContent = 'Reservations (0)';
+      } else {
+        commentsTitle.textContent = `Reservations (${res.length})`;
+        // console.log('---++++', res)
+        let htmlToAppend = '';
+        res.forEach((item) => {
+          const html = `<li>${item.date_start} - ${item.date_end} || by <strong> ${item.username}</strong></li>`;
+          htmlToAppend += html;
+        });
+        const reserveList = document.querySelector('.comment-list');
+        reserveList.innerHTML = htmlToAppend;
+      }
+    }).catch((err) => {
+      commentsTitle.textContent = 'Reservations (0)';
+      throw new Error(err);
+    });
+    commentsTitle.textContent = `Reservations (${reservations === undefined ? '0' : reservations?.length})`;
```

Comment:
- [ ] Good job team on fixing the bug mentioned by the previous reviewer, but there is still one small issue here. when I add a reservation on an album that has 0 reservation before, The reservation gets added but the counter get updated with `Reservation(2)`, then when I add a second reservation it becomes `Reservation(3)`.
Although when I close the popup and then reopen it again, the counter gets corrected, I think it still is a legit bug that you need to sort out.
- [OPTIONAL] Also As you can see in the video below that when I add the first reservation, the `No reservation yet` text still persists there and doesn't go away until I close the popup and reopen it again. I think this bug also needs to be fixed because it was missed by the previous reviewer.



https://user-images.githubusercontent.com/96683843/235302156-c82afd43-d84f-4d88-856c-876d0c9ddf94.mov